solr/core/src/java/org/apache/solr/cli/ExportTool.java

```diff
@@ -660,8 +660,8 @@ class CoreHandler {
       }
 
       boolean exportDocsFromCore() throws IOException, SolrServerException {
-
-        try (SolrClient client = CLIUtils.getSolrClient(baseurl, credentials)) {
+        // reference the replica's node URL, not the baseUrl in scope, which could be anywhere
+        try (SolrClient client = CLIUtils.getSolrClient(replica.getBaseUrl(), credentials)) {
```

Comment:
Since this is a CLI tool, and the CLI tool is potentially being used from a computer outside the cluster itself, I wonder if a side-effect of this might be that it won't work through reverse proxy / ingress with a different external DNS name than what Solr has on the inside.

If I interpret this code correctly, `replica.getBaseUrl()` is data fetched from ZK clusterstate, while `baseUrl` was the user-supplied `--solr-url`. Thus it could be that export tool would currently work through a proxy / ingress due to the auto proxying, but may not work with this change.

Reply:
It's true that this change would break for such a scenario.  Thankfully the ExportTool is using the Solr/HTTP ClusterStateProvider instead of ZK; thus the URLs may be rewritable by such a proxy thing since it'd be at the HTTP level.  But Solr the client would need to be configured to use XML instead of javabin, but I think it's javabin by default.  I could imagine adding special support for this for CLUSTERSTATE or COLSTATUS

Despite that issue, I still believe in this change.  I don't think it makes sense for SolrCloud to go through such guessing efforts to resolve collection/core ambiguity.  Ideally we'd improve on that so that we wouldn't have ambiguity.  I could imagine a /solr/collectionName!shardName/select syntax or similar.

Reply:
It’s a corner case for sure and not worth modifying until there is a user request. Just wanted to highlight the issue and raise awareness as to whether we want bin/solr to work through reverse proxy.